packages/cf.js/test/app-factory.spec.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  NodeMessageType,
+  NodeProposeInstallData
+} from "@counterfactual/node-provider";
+import * as ethers from "ethers";
+
+import { BlockchainAssetType } from "../src/app-types";
+import { Provider } from "../src/provider";
+
+import { TestNodeProvider } from "./fixture";
+
+describe("App Factory", async () => {
+  const TEST_APP_DEFINITION = {
+    address: ethers.constants.AddressZero,
+    appActionEncoding: "tuple(uint256 increase)",
+    appStateEncoding: "tuple(uint256 val)"
+  };
+  let nodeProvider: TestNodeProvider;
+  let provider: Provider;
+
+  beforeEach(() => {
```

Comment:
does this need to be a `beforeEach` or can it just be `beforeAll` / `before`

Reply:
beforeEach is better, makes individual tests idempotent.